standard_sequence/classy_sequence.py

```diff
@@ -765,38 +869,149 @@ def _tweezer_modulation_sequence(self, t):
+                #     shot_globals.mw_biasz_field,
+                # ]
+
+            t = self.BField_obj.ramp_bias_field(
+                    t_aom_off,
+                    bias_field_vector=(mw_biasx_field, mw_biasy_field, mw_biasz_field),
+                    dur=shot_globals.mw_bias_ramp_dur,
+                )
 
-        t = self.TweezerLaser_obj.ramp_power(t, shot_globals.tw_ramp_dur, 1-1e-10)
-        t += 1e-3 # TODO: from the photodetector, the optical pumping beam shutter seems to be closing slower than others
-            # that's why we add extra time here before imaging to prevent light leakage from optical pump beam
 
+            # Making sure there is enough time between the MOT pumping pulse and the killing pulse to switch shutters
+            t = t + max(
+                D2Lasers.CONST_MIN_SHUTTER_ON_TIME
+                + D2Lasers.CONST_SHUTTER_TURN_ON_TIME
+                - shot_globals.tw_ramp_dur,
+                0,
+            )
+            t += shot_globals.mw_field_wait_dur
+            t = self.TweezerLaser_obj.ramp_power(
+                t, shot_globals.tw_ramp_dur, shot_globals.tw_ramp_power
+            )
+
+            if shot_globals.do_mw_pulse:
+                # self.TweezerLaser_obj.aom_off(t)
+                t = self.Microwave_obj.do_pulse(t, shot_globals.mw_time)
+                # self.TweezerLaser_obj.aom_on(t, shot_globals.tw_ramp_power)
+            elif shot_globals.do_mw_sweep:
+                mw_sweep_start = (
+                    shot_globals.mw_detuning + shot_globals.mw_sweep_range / 2
+                )
+                mw_sweep_end = (
+                    shot_globals.mw_detuning - shot_globals.mw_sweep_range / 2
+                )
+                t = self.Microwave_obj.do_sweep(
+                    t, mw_sweep_start, mw_sweep_end, shot_globals.mw_sweep_duration
+                )
+
+            if shot_globals.do_killing_pulse:
+                # If we use the MOT beams for pumping we have to switch shutters
+                # Our pulse function is set so that switching the shutters adds time before the pulse
+                # Here though we want the shutter switching to overlap with the tweezer ramp rather than start after it
+                t = t - D2Lasers.CONST_SHUTTER_TURN_ON_TIME
+                t, _ = self.kill_F4(t, close_all_shutters=False)
+            else:
+                t += shot_globals.op_killing_pulse_time
+
+        if shot_globals.op_label == "sigma":
+            # Making sure the ramp ends right as the pumping is starting
+            t_start_ramp = (
+                t_aom_off - shot_globals.tw_ramp_dur - shot_globals.op_repump_time
+            )
+            _ = self.TweezerLaser_obj.ramp_power(
+                t_start_ramp, shot_globals.tw_ramp_dur, shot_globals.tw_ramp_power
+            )
+            _ = self.TweezerLaser_obj.ramp_power(
+                t_aom_off, shot_globals.tw_ramp_dur, 0.99
+            )
+
```

Comment:
Explain why there are two of these right next to each other. Starting at different times... Would it be more clear to place these in order in the sequence near where they actually happen?

Reply:
Comment added now, if we want the logical to be more clear, it would be possible to split mot op and sigma op into two different function called in main.